### DISCUSSION TITLE
Automatically calculate colsep to fill the entire pagewidth

### DISCUSSION THREAD

**Discussion by note286** [lvjr/tabularray#362] [Ideas]
```latex
\documentclass{article}
\usepackage{showframe}
\usepackage{tabularray}
\begin{document}
\begin{table}
\centering
\caption{title}
\begin{tblr}
{
colspec = {*{6}{l}},
% colsep  = 7.5pt,
colsep  = 0pt,
hlines,vlines
}
Lorem     & ipsum      & dolor        & sit     & amet    & consectetur \\
tempor    & incididunt & ut           & labore  & et      & dolore      \\
quis      & nostrud    & exercitation & ullamco & laboris & nisi        \\
consequat & Duis       & aute         & irure   & dolor   & in          \\
cillum    & dolore     & eu           & fugiat  & nulla   & pariatur    \\
proident  & sunt       & in           & culpa   & qui     & officia     \\
\end{tblr}
\end{table}
\end{document}
```

![image](https://user-images.githubusercontent.com/16207292/216741340-2d3c48fd-8cc9-44b7-acbe-f849eee9c762.png)

In the table above, we set `colsep = 0pt`. I hope tblr can automatically complete the setting of `colsep` so that the table width is the specified width, such as `\textwidth`.

![image](https://user-images.githubusercontent.com/16207292/216741412-fbcc4ece-be9b-4071-82cb-70526b83753e.png)

I try to set `colsep = 7.5pt` to meet the demand in this example. But this is inconvenient and not accurate. Whether there is a method to automatically complete this step. If you can provide an interface, such as `width = \textwidth, colsep = autofill`, it will be even better. 

I know the existence of the `X` column format, but I don't want this. In many cases, the proportion between columns is not easy to see, so it is not very convenient to use the `X` column format. All I want is to set `colsep` automatically.

**Comment by lvjr** [comment]
I have created a new issue #365.
